### PR TITLE
Data type switch

### LINKED ISSUE #33: Plugin: Data Type Switch
AS A: user
I'D LIKE/I NEED: a data type switch in config for container/script plugin types
SO THAT: I can choose if a plugin receives simple command line type data  `bot pager list schedules`

or receive more advanced data in json format such as

``` json
{
  chat: {
    name: username_id,
    channel: channel_from_data,
    timestamp: time_message_was_sent
  },
  command: {
    data: data_text_from_chat
  }
}
```

that can allow a command line tool or app ingest and sort appropriately to provide either API specific response or detailed replies in chat.

### PR DESCRIPTION
**Pull request by TheFynx**
Fixes #33

### CODE REVIEW

**Comment by TheFynx** [comment]
See https://github.com/ImperialLabs/slapin-test/tree/master/advanced for examples on how to utilize the different data type.

File: lib/core/plugin.rb [hunk 1]
```diff
@@ -144,22 +145,29 @@ def help
   # @return string representing response to be displayed
   def exec(data_from_chat = nil)
     # based on some meta information like the type then execute the proper way
+    # Strip an incorrection coded qoutes to UTF8 ones
     data_from_chat.text.tr!('“', '"')
```
Comment (aaronblythe) [inline lib/core/plugin.rb:149]: For completeness should we either replace the other quote and/or log an issue to look into this farther so we don't lose track of it.

**Review by aaronblythe — APPROVED** [review]
Looks straight forward.  Couple minor things, good to go to master.

File: lib/core/plugin.rb (hunk 1 above)
Comment (TheFynx) [reply lib/core/plugin.rb:149]: I'll open an issue for this since it wasn't part of this PR.

**Pull request merged**